File: Leetcode/Daily/24-05-2024/prob.cpp
class Solution {
public:
    int ans = 0;
    int a[26],b[26];
    void f(int ind,vector<string>&words,vector<int>&score){
        if(ind==words.size()){
            int s = 0;
            for(int i=0;i<26;i++){
                if(b[i]>a[i])return;
                s+= score[i]*b[i];
            }

            ans = max(ans,s);
            return;
        }
        for(auto &c:words[ind]){
            b[c-'a']++;
        }
        f(ind+1,words,score);
        for(auto &c:words[ind]){
            b[c-'a']--;
        }

        f(ind+1,words,score);
    }
    int maxScoreWords(vector<string>& words, vector<char>& letters, vector<int>& score) {

        for(auto &c:letters)a[c-'a']++;
        f(0,words,score);

        return ans;
    }
};

File: Leetcode/Daily/24-05-2024/prob.go
var a,b [26]int
var ans int

func f(ind int,words []string,score []int){
    if ind==len(words){
        sum := 0
        for i:=0;i<26;i++{
            if b[i]>a[i]{
                return
            }

            sum += score[i]*b[i]
        }
        ans = max(ans, sum)
        return
    }
    // fmt.Println(ind)
    for _,c := range words[ind]{
        b[c-'a']++
    }
    f(ind+1,words,score)
    for _,c := range words[ind]{
        b[c-'a']--
    }

    f(ind+1,words,score)
}

func maxScoreWords(words []string, letters []byte, score []int) int {
    ans = 0
	for i := range a {
		a[i] = 0
		b[i] = 0
	}

	for _, c := range letters {
		a[c-'a']++
	}

    f(0,words,score)

    return ans
}